patches/minecraft/net/minecraft/core/dispenser/BoatDispenseItemBehavior.java.patch

```diff
@@ -1,12 +1,10 @@
 --- a/net/minecraft/core/dispenser/BoatDispenseItemBehavior.java
 +++ b/net/minecraft/core/dispenser/BoatDispenseItemBehavior.java
-@@ -33,22 +_,54 @@
+@@ -33,22 +_,51 @@
        double d2 = p_123375_.y() + (double)((float)direction.getStepY() * 1.125F);
        double d3 = p_123375_.z() + (double)direction.getStepZ() * d0;
        BlockPos blockpos = p_123375_.getPos().relative(direction);
 +      Boat boat = (Boat)(this.isChestBoat ? new ChestBoat(level, d0, d1, d2) : new Boat(level, d0, d1, d2));
-+      boat.setVariant(this.type);
-+      boat.setYRot(direction.toYRot());
        double d4;
 -      if (level.getFluidState(blockpos).is(FluidTags.WATER)) {
 +      if (boat.canBoatInFluid(level.getFluidState(blockpos))) {
```

Comment:
Couldn't this cause issues for mods overriding the "boat.canBoatInFluid", since the Variant is potentially incorrect?